_posts/en/posts/2024-06-10-disclose-unbounded-banlist.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Disclosure: CPU/memory DoS due to many malicious peers (<= version 0.20.0)
+name: blog-disclose-unbounded-banlist
+id: en-blog-disclose-unbounded-banlist
+lang: en
+type: advisory
+layout: post
+
+## If this is a new post, reset this counter to 1.
+version: 1
+
+## Only true if release announcement or security annoucement. English posts only
+announcement: 1
+
+excerpt: >
+  Public disclosure of a DoS vulnerability affecting old versions of Bitcoin Core
+---
+
+Bitcoin Core maintained an unlimited list of banned IP addresses and performed a quadratic operation
+on it. This could lead to an OOM crash and a CPU Dos.
+
+This issue is considered **High** severity.
+
+## Details
+
+Bitcoin Core maintained a list of banned IP addresses. This list was not bounded and could be
+manipulated by an adversary. Adding new entries to this list was particularly cheap for an attacker
+when considering IPV6. In addition, when receiving a `GETADDR` message, Bitcoin Core would scan the
+entire ban list for every single address to be returned (up to 2500).
+
+## Attribution
+
+Calin Culianu first responsibly disclosed it. Calin later publicly disclosed the bug in [a PR
+comment](https://github.com/bitcoin/bitcoin/pull/15617#issuecomment-640898523).
+
+On the same day Jason Cox from Bitcoin ABC emailed the Bitcoin Core project to share this same
+report they also received.
+
+## Timeline
+
+- 2020-06-08 Calin Culianu privately reports the bug to the Bitcoin Core project
+- 2020-06-08 Jason Cox privately shares the (same) report sent to Bitcoin ABC with Bitcoin Core
+- 2020-06-08 Calin Culianu publicly discloses the vulnerability on the original PR which introduced the quadratic behaviour
+- 2020-06-09 Pieter Wuille opens PR [#19219](https://github.com/bitcoin/bitcoin/pull/19219) which fixes both the unbounded memory usage and the quadratic behaviour
+- 2020-06-16 Luke Dashjr gets assigned CVE-2020-14198 for this vulnerability after his request
+- 2020-07-07 Pieter's PR is merged
+- 2020-08-01 Bitcoin Core 0.20.1 is released with the fix
+- 2021-01-14 Bitcoin Core 0.21.0 is released with the fix
+- 2022-04-25 The last vulnerable Bitcoin Core version (0.20.0) goes EOL
```

Comment:
According to old versions of the old lifecycle table (which were used to create the new table), v0.20.0 went end-of-life on 2022-02-01. One of the two should probably be amended.

Corollary: Should the EOL always be based on the release date of the Major version three later or the target release date of the Major version three later?

Reply:
According to https://github.com/bitcoin/bitcoin/releases/tag/v23.0 and the dates at https://bitcoincore.org/bin/bitcoin-core-23.0/ 23.0 was released on April 25th. So i think the date here is correct.

> Corollary: Should the EOL always be based on the release date of the Major version three later or the target release date of the Major version three later?

No strong opinion but i had assumed it's when the release 3 versions later actually get released.